pkg/querier/querier.go

```diff
@@ -382,6 +387,16 @@ func (mq multiQuerier) Select(ctx context.Context, _ bool, sp *storage.SelectHin
 		return storage.ErrSeriesSet(NewMaxQueryLengthError(endTime.Sub(startTime), maxQueryLength))
 	}
 
+	// If the original request didn't have a limit but the limit was enforced, annotate the resulting series set with a warning.
+	if sp.Limit == 0 && sp.Limit != limit {
+		defer func() {
+			var warning annotations.Annotations
+			warning.Add(errors.New("results may be truncated due to limit"))
```

Comment:
can we add more details - what is the limit and what was the requested limit

Reply:
This only adds a warning when the request didn't send a limit. My thinking was that when I pass a `?limit=` explicitly, Prometheus adds its own warning to the response (ref [code](https://github.com/grafana/mimir/blob/9dd7b61a2bf7f2755cda179bf6ad2251d98ade77/vendor/github.com/prometheus/prometheus/web/api/v1/api.go#L982)).

Reply:
oh, i misread `&&` as `||`, so I assumed this will add a warning when we clamp the limit. Do you want to do that too?

Reply:
I'm not sure. We don't do anything like that when we clamp other request parameters. I'm not sure if we should do that for the `limit`. So I thought to only warn the user when they didn't pass the limit, but the limit was enforced, as this is the most unexpected scenario for them.

Reply:
Could you take another a look. In the https://github.com/grafana/mimir/pull/10748/commits/50eaaf85eddc34852d0c11ea7888d517893794c0 I've added a more detailed warning to the response.